### PR TITLE
Stats: Fix requests to have the http envelope

### LINKED ISSUE #57775: Stats: Ads section empty
### Quick summary

Right now it seems that any site's stats display may be completely blank for the ads section if they have WordAds enabled and are receiving impressions.

### Steps to reproduce

1. Navigate to My Site > Stats > Ads
2. Check for some data in the charge

### What you expected to happen

To see some data, even if it says zero

### What actually happened

It shows completely blank, even if there are impressions and earnings on the site.

Screen Shot 2021-11-08 at 1 13 13 PM

### Context

reported by customers in 

32630437-hc
27510796-hc
32629491-hc

Reproducible on any site I have right now

### Operating System

_No response_

### Browser

_No response_

### Simple, Atomic or both?

Simple, Atomic

### Theme-specific issue?

_No response_

### Other notes

_No response_

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

_No response_

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR fixes a bug we introduced in #57518 where we didn't specify requests to have the HTTP envelope.

Fixes #57775.

#### Testing instructions

* Go to `/stats/ads/year/:site` where `:site` has WordAds enabled.
* Verify stats appear properly now.

### CODE REVIEW

File: client/state/stats/lists/actions.js [hunk 1]
```diff
@@ -66,7 +66,6 @@ export function requestSiteStats( siteId, statType, query ) {
 		let apiNamespace;
 		if ( wpcomV1Endpoints.hasOwnProperty( statType ) ) {
 			subpath = wpcomV1Endpoints[ statType ];
-			apiNamespace = 'rest/v1.1';
```
Comment (jsnajdr) [inline client/state/stats/lists/actions.js:69]: I told you so! 😄 Apparently it was not just a style nit, but the mere presence of `apiNamespace` means we are in WP REST API territory and that the envelope query param is different (WP.com `http_envelope` vs WP `_envelope`).

Comment (tyxla) [reply client/state/stats/lists/actions.js:69]: I stand corrected! I'll just admit and pretend this behavior makes sense in the first place 😆

**Review by jsnajdr — APPROVED** [review]
:+1:

File: client/state/stats/lists/actions.js (hunk 1 above)
Comment (fullofcaffeine) [reply client/state/stats/lists/actions.js:69]: Interesting, what was the original rationale for adding this line, and why does removing it fixes the envelope query param (as described in the PR's title)?

Comment (tyxla) [reply client/state/stats/lists/actions.js:69]: Good question @fullofcaffeine, let me add some context.

So, this line was added in #57518 as part of our work to refactor away from `wpcom.undocumented()` methods. 

As you can see, in [the old code](https://github.com/Automattic/wp-calypso/pull/57518/files#diff-c3e509bb1fa8af9056af182e044d339272e486c7911f0fa3bb82695b8b211654L53), we didn't specify `apiNamespace` specifically. 

However, in the new version, we did. And if you take a look at the library that performs the requests, we'll notice that the HTTP envelope argument [depends on the presence of `isRestApi`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom-xhr-request/src/index.js#L266), which [itself depends on](https://github.com/Automattic/wp-calypso/blob/trunk/packages/wpcom-xhr-request/src/index.js#L216) whether `apiNamespace` has been **defined**. So by defining `apiNamespace`, we were no longer using the HTTP envelope. 

Now, you might ask, why does the HTTP envelope change anything? Well, it alters the behavior of the endpoint response in a pretty significant way. It will change the endpoint to always send HTTP 200 (regardless of the outcome) and will wrap the JSON output in a wrapper containing the "real" HTTP status code and some of the headers. This was useful historically because many clients could potentially block non-200 status code responses. 

So, to summarize, in our case, the client was expecting the response to be in the format that's provided by an HTTP-envelope-wrapped endpoint, and by declaring `apiNamespace`, we were technically removing the envelope, rendering the output in a format that's unexpected by the client.

Let me know if that makes sense 😉

Comment (jsnajdr) [reply client/state/stats/lists/actions.js:69]: > So by defining `apiNamespace`, we were no longer using the HTTP envelope.

More precisely, we were using wrong query param to request the envelope. The old v1 API uses `http_envelope`, the new v2 uses `_envelope`. When `apiNamespace` is specified, the library assumes v2, even when we passed a v1 namespace string.
